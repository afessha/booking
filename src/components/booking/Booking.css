body {
  padding-top: 5px;
  padding-bottom: 10px;
  background-color: #eee;
}

.form-signin {
  max-width: 330px;
  padding: 10px;
  margin: 0 auto;
  text-align: center;
}
.form-signin .form-signin-heading {
  margin-bottom: 10px;
  color: white;
}
.container {
  background-color: #122f4c;
  padding: 0;
}

.form-signin label {
  color: white;
  font-size: 18px;
}
.form-signin input {
  border-radius: 2px;
}

.form-signin input {
  margin-bottom: 10px;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.footer {
  margin: 10px;
  text-align: center;
  color: #122f4c;
  font-weight: bold;
}
.error {
  color: red;
}
.submitted {
  color: green;
}
