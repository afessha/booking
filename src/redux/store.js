import { createStore } from "redux";
import bookingReducer from "./reducer";

const store = createStore(
  bookingReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
