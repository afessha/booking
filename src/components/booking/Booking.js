import React from "react";
import { connect } from "react-redux";
import axios from "axios";
import "./Booking.css";
import Footer from "../footer/Footer";
import { useFormik } from "formik";
import * as Yup from "yup";
import { bookingAdd } from "../../redux/actions";

function Booking(props) {
  const formik = useFormik({
    initialValues: {
      firstname: "",
      lastname: "",
      dining_date: "",
      number_of_covers: "",
      booking_email: "",
      phone_number: "",
      status: "Not arrived",
    },
    validationSchema: Yup.object({
      firstname: Yup.string()
        .max(15, "Must be 15 characters or less")
        .required("Required"),
      lastname: Yup.string()
        .max(20, "Must be 20 characters or less")
        .required("Required"),
      booking_email: Yup.string()
        .email("Invalid email address")
        .required("Required"),
    }),
    onSubmit: async (booking) => {
      try {
        let res = await axios.post(
          "https://a6qroxzfnb.execute-api.eu-west-2.amazonaws.com/dev/bookings",
          JSON.stringify(booking)
        );
        console.log(res.status);
        if (res.status === 200) {
          props.addNew({ booking });
          formik.resetForm();
        }
      } catch (err) {
        console.log("Error happened" + err);
      }
    },
  });

  return (
    <>
      <div>
        <div className="container">
          <form className="form-signin" onSubmit={formik.handleSubmit}>
            <h2 className="form-signin-heading">Booking</h2>
            <br />
            <label>First Name</label>
            <input
              className="form-control"
              name="firstname"
              {...formik.getFieldProps("firstname")}
            />
            {formik.touched.firstname && formik.errors.firstname ? (
              <div className="error">{formik.errors.firstname}</div>
            ) : null}
            <label>Last Name</label>
            <input
              className="form-control"
              name="lastname"
              {...formik.getFieldProps("lastname")}
            />
            {formik.touched.lastname && formik.errors.lastname ? (
              <div className="error">{formik.errors.lastname}</div>
            ) : null}
            <label> Dining Date</label>
            <input
              className="form-control"
              name="dining_date"
              {...formik.getFieldProps("dining_date")}
            />
            <label>Number of Covers </label>
            <input
              className="form-control"
              name="number_of_covers"
              {...formik.getFieldProps("number_of_covers")}
            />

            <label> Email </label>
            <input
              className="form-control"
              name="booking_email"
              {...formik.getFieldProps("booking_email")}
            />
            {formik.touched.booking_email && formik.errors.booking_email ? (
              <div className="error">{formik.errors.booking_email}</div>
            ) : null}
            <label>Phone Number</label>
            <input
              className="form-control"
              name="phone_number"
              {...formik.getFieldProps("phone_number")}
            />
            <button className="btn btn-lg btn-success btn-block" type="submit">
              Book Table
            </button>
          </form>
        </div>
        <Footer />
      </div>
    </>
  );
}

const mapDispatchToProps = (dispatch) => {
  return {
    addNew: (booking) => dispatch(bookingAdd(booking)),
  };
};
export default connect(null, mapDispatchToProps)(Booking);
