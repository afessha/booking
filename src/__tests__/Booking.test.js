import React from "react";
import { create } from "react-test-renderer";
import Booking from "../components/booking/Booking";
import { Provider } from "react-redux";
import store from "../redux/store";

describe("Booking component", () => {
  test("Matches the snapshot", () => {
    const booking = create(
      <Provider store={store}>
        <Booking />
      </Provider>
    );
    expect(booking.toJSON()).toMatchSnapshot();
  });
});
