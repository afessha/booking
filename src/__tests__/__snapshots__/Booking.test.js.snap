// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking component Matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <form
            class="form-signin"
          >
            <h2
              class="form-signin-heading"
            >
              Booking
            </h2>
            <br />
            <label>
              First Name
            </label>
            <input
              class="form-control"
              name="firstname"
              value=""
            />
            <label>
              Last Name
            </label>
            <input
              class="form-control"
              name="lastname"
              value=""
            />
            <label>
               Dining Date
            </label>
            <input
              class="form-control"
              name="dining_date"
              type="date"
              value=""
            />
            <label>
              Number of Covers 
            </label>
            <input
              class="form-control"
              name="number_of_covers"
              type="number"
              value=""
            />
            <label>
               Email 
            </label>
            <input
              class="form-control"
              name="booking_email"
              value=""
            />
            <label>
              Phone Number
            </label>
            <input
              class="form-control"
              name="phone_number"
              value=""
            />
            <button
              class="btn btn-lg btn-success btn-block"
              type="submit"
            >
              Book Table
            </button>
          </form>
        </div>
        <div
          class="footer"
        >
           Awet Fessha ©2020 
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <form
          class="form-signin"
        >
          <h2
            class="form-signin-heading"
          >
            Booking
          </h2>
          <br />
          <label>
            First Name
          </label>
          <input
            class="form-control"
            name="firstname"
            value=""
          />
          <label>
            Last Name
          </label>
          <input
            class="form-control"
            name="lastname"
            value=""
          />
          <label>
             Dining Date
          </label>
          <input
            class="form-control"
            name="dining_date"
            type="date"
            value=""
          />
          <label>
            Number of Covers 
          </label>
          <input
            class="form-control"
            name="number_of_covers"
            type="number"
            value=""
          />
          <label>
             Email 
          </label>
          <input
            class="form-control"
            name="booking_email"
            value=""
          />
          <label>
            Phone Number
          </label>
          <input
            class="form-control"
            name="phone_number"
            value=""
          />
          <button
            class="btn btn-lg btn-success btn-block"
            type="submit"
          >
            Book Table
          </button>
        </form>
      </div>
      <div
        class="footer"
      >
         Awet Fessha ©2020 
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
